crmsh/report/core.py

```diff
@@ -441,6 +432,31 @@ def load_context_attributes(context: Context) -> None:
     context.cores_dir_list.extend([constants.COROSYNC_LIB] if os.path.isdir(constants.COROSYNC_LIB) else [])
 
 
+def load_context_trace_dir_list(context: Context) -> None:
+    log_contents = ""
+    cmd = f"grep 'INFO: Trace for .* is written to ' {log.CRMSH_LOG_FILE}*|grep -v 'collect'"
+    for node in context.node_list:
+        shell = context.passwordless_shell_for_nodes.get(node)
+        if shell is not None:
+            ret = shell.subprocess_run_without_input(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        else:
+            # This is a last effort when passwordless ssh is not available
+            node = f"{context.ssh_user}@{node}" if context.ssh_user else node
+            logger.info('Creating ssh connection to %s...', node)
+            cmd = cmd.replace('"', '\\"')
+            cmd = 'ssh {} {} "{}{}"'.format(
+                constants.SSH_OPTS,
+                node,
+                '' if context.ssh_user is None or context.ssh_user == 'root' else 'sudo ',
+                cmd,
+            )
+            shell = crmsh.sh.LocalShell()
+            ret = shell.su_subprocess_run(None, cmd, tty=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Is the same code as line 264-275?

Reply:
Extracted as a function `_interactive_ssh_run_as_root`.